spec/support/ruby_syntax.rb

```diff
@@ -0,0 +1,7 @@
+require 'ripper'
+
+RSpec::Matchers.define :be_valid_as_ruby do
+  match do |actual|
+    !!Ripper.sexp(actual)
+  end
+end
```

Comment:
I added a new matcher to test the output is valid as a Ruby program. Because I have no confidence to write correct generator of Ruby program.